pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -49,6 +49,8 @@ type framework struct {
 	postBindPlugins       []PostBindPlugin
 	unreservePlugins      []UnreservePlugin
 	permitPlugins         []PermitPlugin
+	plugins               *config.Plugins
```

Comment:
instead of adding these here (and never GC the config), they should go in `factory.Config`. They are only useful for `factory_test.go` anyways.

Reply:
Aren't we planning to remove that in the future?

Reply:
that practically means we need to write a function for the framework to re-create the config from the plugin arrays, I don't want to do that :) 

I feel factory.Config already has too many arguments, adding more just makes it more cluttered and confusing.

Reply:
> I feel factory.Config already has too many arguments, adding more just makes it more cluttered and confusing.

I agree. My only point is that configuration should only exist at startup and be collected after that.

Reply:
> > I feel factory.Config already has too many arguments, adding more just makes it more cluttered and confusing.
> 
> I agree. My only point is that configuration should only exist at startup and be collected after that.

I am not disagreeing with that, I am just saying that while it is more appropriate to have it in factory.Config (because it should exist only at startup), that place is cluttered and has the potential to get deprecated. That being said, it does not matter too much to me, if you feel strongly about having it there, I can move it there.

Reply:
Even if the `Config` struct goes away, it looks like we will preserve some utility functions. Then the test could be for them.

Reply:
but right now it is very cluttered, there are so many things called "config" there.

Reply:
what will add to the confusion is that in that package we store and pass around plugins/pluginConfig that we received from component config (which is a subset of this one).

Reply:
I think naming is less of a problem than having living objects that we do not care about at runtime.
For now, this is my suggestion:
In `Config` we have a struct named `Framework` that stores the parameters to `NewFramework`. That is implicitly a configuration because it lives inside the `Config` struct. The instantiation of the framework itself is moved to `scheduler.NewFromConfig`.

Reply:
I don't think having the Config here is a big problem, they are implementations details of the framework anyways. However, I feel a bit hesitant on adding the `GetConfiguration` to the Framework interface, which could open the door for changing the config after framework initialization. Having said that, I don't think it's too big an issue.

One way to get rid of this is to test the unexported `factory` method of `getPriorityConfigs` and `getPredicateConfigs`, but I prefer to test exported methods.

Ideally we can avoid the new  `GetConfiguration`  interface method but I don't hold strongly.

Reply:
we can't do the instantiation there because the framework is needed when making generic scheduler. I will move those to Config though.

Reply:
Sg. FYI @ahmad-diaa for when you work on this refactoring.

Reply:
done.